File: backend/schemas/authSchemas.py
from pydantic import BaseModel, EmailStr
from typing import Optional

class LoginInput(BaseModel):
    email: EmailStr
    senha: str

class Token(BaseModel):
    access_token: str
    token_type: str = "bearer"

class TokenData(BaseModel):
    sub: str  
    email: EmailStr
    exp: Optional[int] = None

File: backend/schemas/authSchemas.js
const Joi = require('joi');

// Schema for LoginInput
const LoginInputSchema = Joi.object({
  email: Joi.string().email().required(),
  senha: Joi.string().required(),
});

// Schema for Token
const TokenSchema = Joi.object({
  access_token: Joi.string().required(),
  token_type: Joi.string().default('bearer'),
});

// Schema for TokenData
const TokenDataSchema = Joi.object({
  sub: Joi.string().required(),
  email: Joi.string().email().required(),
  exp: Joi.number().optional(),
});

module.exports = {
  LoginInputSchema,
  TokenSchema,
  TokenDataSchema,
};